cmake/sysbuild/b0_mcuboot_signing.cmake

```diff
@@ -69,7 +69,7 @@ function(ncs_secure_boot_mcuboot_sign application bin_files signed_targets prefi
         # Hence, if a programmer is given this hex file, it will flash it
         # to the secondary slot, and upon reboot mcuboot will swap in the
         # contents of the hex file.
-        ${imgtool_sign} ${PROJECT_BINARY_DIR}/signed_by_b0_${application}.bin ${output}.bin
+        ${imgtool_sign} ${imgtool_extra} ${PROJECT_BINARY_DIR}/signed_by_b0_${application}.bin ${output}.bin
```

Comment:
the use of `imgtool_extra` is actually problematic because it will inherit `imgtool_extra` from parent scope value which can result in surprises.

Use of imgtool_extra should be an argument to the function here:
https://github.com/nrfconnect/sdk-nrf/blob/f30dcc20c0860d962676b0f54f9b7c27816c2ddd/cmake/sysbuild/b0_mcuboot_signing.cmake#L13

Reply:
it won't inherit anything because this is from a sysbuild context, there is no imgtool_extra variable in sysbuild